stream-chat-android-client/src/main/java/io/getstream/chat/android/client/channel/ChannelClient.kt

```diff
@@ -267,6 +268,19 @@ public class ChannelClient internal constructor(
         return client.sendMessage(channelType, channelId, message)
     }
 
+    /**
+     * Sends the message to the given channel without running any side effects.
+     *
+     * @param message Message to send.
+     *
+     * @return Executable async [Call] responsible for sending a message.
+     */
+    @InternalStreamChatApi
+    @CheckResult
+    public fun sendMessageInternal(message: Message): Call<Message> {
```

Comment:
Do we really need this? I think we can refactor the code in `retryMessagesWithSyncedAttachments` and `MessageSendingService.doSend` and remove it.